roles/installer/templates/postgres.yaml.j2

```diff
@@ -37,10 +37,27 @@ spec:
       imagePullSecrets:
         - name: {{ image_pull_secret }}
 {% endif %}
+      initContainers:
+        - name: init-chmod-data
+          image: '{{ postgres_image }}:{{ postgres_image_version }}'
+          imagePullPolicy: '{{ image_pull_policy }}'
+          command:
+            - /bin/sh
+            - -c
+            - |
+              if [ ! -f {{ postgres_data_path }}/PG_VERSION ]; then
+                chown postgres:root {{ postgres_data_path | dirname }}
+              fi
+          volumeMounts:
+            - name: postgres
+              mountPath: '{{ postgres_data_path | dirname }}'
+              subPath: '{{ postgres_data_path | dirname | basename }}'
       containers:
         - image: '{{ postgres_image }}:{{ postgres_image_version }}'
           imagePullPolicy: '{{ image_pull_policy }}'
           name: postgres
+          securityContext:
+            fsGroup: 999
```

Comment:
@rooftopcellist  I guess we will need to modify this to match the UID for the `rhel8/postgresql-12` so the approach can work on both scenarios.

Reply:
*postgres:12*

```
postgres@add799ebb405:/$ id
uid=999(postgres) gid=999(postgres) groups=999(postgres),101(ssl-cert)
```

*rhel-8/postgresql-12*

```
bash-4.4$ id
uid=26(postgres) gid=26(postgres) groups=26(postgres),0(root)
```

As you mentioned, we will need to conditionally set the fsGroup to cover this case.